src/content/containers/TimelineMarkers.js

```diff
@@ -0,0 +1,168 @@
+    const height = this.getViewHeight(maxViewportHeight);
+    const buttonClass = classNames('timelineMarkersCollapseButton', {
+      expanded: isRowExpanded,
+      collapsed: !isRowExpanded,
+    });
+
+    return (
+      <div className='timelineMarkers' style={{ height }}>
+        <div className='timelineMarkersLabels grippy' title={processDetails}>
+          <span>{threadName}</span>
+          <button className={buttonClass} onClick={this.toggleThreadCollapse} />
+        </div>
+        <TimelineMarkerCanvas key={threadIndex}
+                            // TimelineViewport props
+                            isRowExpanded={isRowExpanded}
+                            isSelected={isSelected}
+                            timeRange={timeRange}
+                            maxViewportHeight={maxViewportHeight}
+                            getScrollElement={getScrollElement}
+                            maximumZoom={this.getMaximumZoom()}
+                            selection={selection}
+                            updateProfileSelection={updateProfileSelection}
+                            viewportNeedsUpdate={(prevProps, newProps) => {
+                              return prevProps.markerTimingRows !== newProps.markerTimingRows;
+                            }}
+
+                            // TimelineMarkerCanvas props
+                            interval={interval}
+                            thread={thread}
+                            rangeStart={timeRange.start}
+                            rangeEnd={timeRange.end}
+                            markerTimingRows={markerTimingRows}
+                            getCategory={getCategory}
+                            getLabel={getLabel}
+                            maxMarkerRows={maxMarkerRows}
+                            markers={markers}
+                            rowHeight={ROW_HEIGHT} />
+      </div>
+    );
+  }
+}
+
+
+
+export default connect((state, ownProps) => {
+  const { threadIndex } = ownProps;
+  const threadSelectors = selectorsForThread(threadIndex);
+  const isRowExpanded = getAreMarkersExpanded(state, threadIndex);
+
+  const markers = threadSelectors.getTracingMarkers(state);
+  const markerTimingRows = isRowExpanded
+    ? threadSelectors.getMarkerTiming(state)
+    : [];
+
+  return {
+    thread: threadSelectors.getFilteredThreadForFlameChart(state),
+    isRowExpanded,
+    markers,
+    markerTimingRows,
+    maxMarkerRows: markerTimingRows.length,
+    isSelected: true,
+    timeRange: getDisplayRange(state),
+    interval: getProfileInterval(state),
+    getCategory: getCategoryColorStrategy(state),
+    getLabel: isRowExpanded ? getLabelingStrategy(state) : getImplementationName,
+    threadIndex,
+    selection: getProfileViewOptions(state).selection,
+    threadName: threadSelectors.getFriendlyThreadName(state),
+    processDetails: threadSelectors.getThreadProcessDetails(state),
+  };
+}, (actions: Object))(TimelineMarkers);
```

Comment:
I believe so because of the use of `changeTimelineMarkersExpandedThread`

```js
  _toggleThreadCollapse() {
    const { changeTimelineMarkersExpandedThread, threadIndex, isRowExpanded } = this.props;
    changeTimelineMarkersExpandedThread(threadIndex, !isRowExpanded);
  }
```